### PR TITLE
Several improvements

### PR DESCRIPTION
**Pull request by ruimarinho**
- Added function name to improve debugging.
- Added support for exposing the header on the response.
- Added support to skip header lookup.
- Added support to skip querystring lookup.
- Added support for customising request header and querystring parameter names.
- Uses recommended `ctx.state` on newer version of `koa` (added BC support and tests).
- Updated dependencies
- Updated .travis.yml to test on node 0.12 and io.js.

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -1,3 +1,8 @@
 language: node_js
+
 node_js:
   - "0.11"
+  - "0.12"
+  - iojs
+
+sudo: false
```
Comment (juliangruber) [inline .travis.yml:8]: what does this line mean?

Comment (ruimarinho) [inline .travis.yml:8]: It allows Travis to use the container-based infrastructure for faster build times.

**Comment by ruimarinho** [comment]
@juliangruber any chance of taking at look at the PR?

**Comment by ruimarinho** [comment]
@juliangruber removed `should` in favor of `assert`.

**Comment by ruimarinho** [comment]
Is there anything missing from the PR that I can help with?

**Comment by ruimarinho** [comment]
ping @juliangruber

**Comment by ruimarinho** [comment]
It looks like this library is not being actively maintained anymore so I've published these and other improvements as part of https://www.npmjs.com/package/koa-requestid.

**Pull request closed without merging**